src/components/EmbedMode.js

```diff
@@ -1,17 +1,24 @@
 import React, { useMemo } from "react";
+import { useLocation } from "react-router";
+import { URLSearchParams as NodeURLSearchParams } from "url";
 
-const DEFAULTS = { enabled: false, settings: {} };
+const EmbedModeContext = React.createContext({ enabled: false, settings: {} });
 
-const EmbedModeContext = React.createContext(DEFAULTS);
+function getUrlSearchParams(search) {
+    if (typeof window === "undefined") {
+        return new NodeURLSearchParams(search);
+    } else {
+        return new window.URLSearchParams(search);
+    }
+}
```

Comment:
`ExecutionEnvironment.canUseDOM` seems to be the idiomatic Docusaurus way.

```suggestion
function getUrlSearchParams(search) {
    if (ExecutionEnvironment.canUseDOM) {
        return new URLSearchParams(search);
    } else {
        return new NodeURLSearchParams(search);
    }
}
```

Reply:
Thanks for the investigation. I resorted to just `search.includes("embed")`, but I'm thinking soon we'll have more params and we'll need something anyway. Maybe import some light library that can parse queries isomorphically.